packages/core/content-type-builder/admin/src/components/DataManagerProvider/utils/retrieveNestedComponents.ts

```diff
@@ -1,24 +1,60 @@
-import { makeUnique } from '../../../utils/makeUnique';
+import type { Components, AttributeType } from '../../../types';
+import type { Internal } from '@strapi/types';
 
-export const retrieveNestedComponents = (appComponents: any) => {
-  const nestedComponents = Object.keys(appComponents).reduce((acc: any, current) => {
-    const componentAttributes = appComponents?.[current]?.schema?.attributes ?? [];
-    const currentComponentNestedCompos = getComponentsFromComponent(componentAttributes);
+export type NestedComponent = {
+  component: Internal.UID.Component;
+  uidsOfAllParents?: Internal.UID.Component[];
+  parentCompoUid?: Internal.UID.Component;
+};
 
+export const retrieveNestedComponents = (appComponents: Components): NestedComponent[] => {
+  const nestedComponents = Object.keys(appComponents).reduce((acc: NestedComponent[], current) => {
+    const componentAttributes = appComponents?.[current]?.schema?.attributes ?? [];
+    const currentComponentNestedCompos = getComponentsFromComponent(
+      componentAttributes,
+      current as Internal.UID.Component
+    );
     return [...acc, ...currentComponentNestedCompos];
   }, []);
 
-  return makeUnique(nestedComponents);
+  return mergeComponents(nestedComponents);
 };
 
-const getComponentsFromComponent = (componentAttributes: any) => {
-  return componentAttributes.reduce((acc: any, current: any) => {
+const getComponentsFromComponent = (
```

Comment:
Some docs / a new name for this fucntion would be nice I think. Naming is tricky
though if I understand it right it's kinda:

getNestedComponents? getComponentsNestedWithinComponent? 😄

Reply:
yeah `getComponentsNestedWithinComponent` could work well